nucliadb_node/src/shards/shard_reader.rs

```diff
@@ -474,6 +527,59 @@ impl ShardReader {
         })
     }
 
+    #[tracing::instrument(skip_all)]
+    pub fn download_file_iterator(
+        &self,
+        relative_path: String,
+    ) -> NodeResult<ShardFileChunkIterator> {
+        let span = tracing::Span::current();
+        run_with_telemetry(info_span!(parent: &span, "download file iteration"), || {
+            ShardFileChunkIterator::new(self.root_path.join(relative_path), CHUNK_SIZE)
+        })
+    }
+
+    fn visit_directories(
+        &self,
+        path: PathBuf,
+        to_visit: &mut Vec<PathBuf>,
+    ) -> NodeResult<Vec<ShardFile>> {
+        let dir = fs::read_dir(path)?;
+        let mut files = Vec::new();
+
+        for child in dir {
+            let child = child?;
+            if child.metadata()?.is_dir() {
+                to_visit.push(child.path());
+            } else if let Ok(metadata) = child.metadata() {
+                let path = child.path().into_os_string().into_string().unwrap();
+                if let Some(path) = path.strip_prefix(&self.suffixed_root_path) {
```

Comment:
I personally try to avoid too much nesting, I see you are using `child.metadata()` differently in both branches of the control flow. Maybe you could:
```rust
for child in dir {
     let child = child?;
     let child_metadata = child.metadata()?;
     let Ok(child_path) = child.path().into_os_string().into_string() else {
             continue;
     };
     if child_metada.is_dir() {
            to_visit.push(child.path());
     } else if Some(path) = path.strip_prefix(&self.suffixed_root_path) {
            files.push(ShardFile {
                   relative_path: path.to_string(),
                   size: metadata.len(),
             });
     }
```